Library/Homebrew/cask/cmd/upgrade.rb

```diff
@@ -40,68 +40,81 @@ def run
         end
         puts cask_upgrades.join(", ")
 
+        caught_exceptions = []
         outdated_casks.each do |old_cask|
-          odebug "Started upgrade process for Cask #{old_cask}"
-          raise CaskUnavailableError.new(old_cask, "The Caskfile is missing!") if old_cask.installed_caskfile.nil?
+          begin
+            upgrade_cask(old_cask)
+          rescue CaskError => e
+            caught_exceptions << e
+            next
+          end
+        end
+        return if caught_exceptions.empty?
+        raise caught_exceptions[0] if caught_exceptions.length == 1
+        raise MultipleCaskErrors, caught_exceptions if caught_exceptions.length > 1
+      end
 
-          old_cask = CaskLoader.load(old_cask.installed_caskfile)
+      def self.help
+        "upgrades all outdated casks"
+      end
 
-          old_config = old_cask.config
+      def upgrade_cask(old_cask)
+        odebug "Started upgrade process for Cask #{old_cask}"
+        raise CaskUnavailableError.new(old_cask, "The Caskfile is missing!") if old_cask.installed_caskfile.nil?
 
-          old_cask_installer =
-            Installer.new(old_cask, binaries: binaries?,
-                                    verbose:  verbose?,
-                                    force:    force?,
-                                    upgrade:  true)
+        old_cask = CaskLoader.load(old_cask.installed_caskfile)
```

Comment:
Why reload it, if we already get it as an argument?

Reply:
Not sure. I didn't actually change any of the code in `upgrade_cask`, just moved it out of a loop and into it's own method. Should I remove it?

Reply:
So it looks like the cask object loaded here is actually different from the cask object passed in as a parameter, but this cask is actually loaded earlier in the parent method for the purpose of printing out "old version -> new version". 

I'll try to change it so it doesn't get called twice.